src/platform/tests/TestPlatformMgr.cpp

```diff
@@ -44,15 +44,105 @@ using namespace chip::DeviceLayer;
 //      Unit tests
 // =================================
 
-static void TestPlatformMgr_Init(nlTestSuite * inSuite, void * inContext)
+static void TestPlatformMgr_InitShutdown(nlTestSuite * inSuite, void * inContext)
 {
     CHIP_ERROR err = PlatformMgr().InitChipStack();
     NL_TEST_ASSERT(inSuite, err == CHIP_NO_ERROR);
+
+    err = PlatformMgr().Shutdown();
+    NL_TEST_ASSERT(inSuite, err == CHIP_NO_ERROR);
+}
+
+static void TestPlatformMgr_BasicEventLoopTask(nlTestSuite * inSuite, void * inContext)
+{
+    CHIP_ERROR err = PlatformMgr().InitChipStack();
+    NL_TEST_ASSERT(inSuite, err == CHIP_NO_ERROR);
+
+    err = PlatformMgr().StartEventLoopTask();
```

Comment:
Is there any way to actually prove from a testability standpoint the difference in behavior here vs. calling `RunEventLoop`? 

i.e show that it returns, but still continues to do work on a separable execution context.

e.g

```
 CHIP_ERROR err = PlatformMgr().InitChipStack();
 NL_TEST_ASSERT(inSuite, err == CHIP_NO_ERROR);

PlatformMgr().ScheduleWork(DidWorkFunc);

NL_TEST_ASSERT(inSuite, DidWorkCheck == false);

PlatformMgr().StartEventLoopTask();

// setup a cond var that gets signaled by the function above, but times out after a while

err = PlatformMgr().StopEventLoopTask();

...
```

Reply:
Yes, we could probably do something like that.  I ran out of time to add more interesting tests here this morning, but would be very much in favor of more of that happening!